### PR TITLE
[AIP-135] Document all AIP-135 rules.

### PR DESCRIPTION
**Pull request by lukesneeringer**

### CODE REVIEW

File: docs/rules/core/0135-request-message-name.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  option (google.api.http) = {
+    get: "/v1/{name=publishers/*/books/*}"
+  };
+}
+```
+
+If you need to violate this rule for an entire file, place the comment at the
+top of the file.
+
+[aip-135]: https://aip.dev/135
```
[the remark this answers is not in the document]
Comment (lukesneeringer) [reply docs/rules/core/0135-request-message-name.md]: No, and prettier (the formatting tool) forces them all to lower case.

File: docs/rules/core/0135-request-name-field.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+rule:
+  aip: 135
+  name: [core, '0135', request-name-field]
+  summary: Delete RPCs must have a `name` field in the request.
+---
+
+# Delete methods: Name field
+
+This rule enforces that all `Delete` standard methods have a `string name`
+field in the request message, as mandated in [AIP-135](http://aip.dev/135).
```
[the remark this answers is not in the document]
Comment (lukesneeringer) [reply docs/rules/core/0135-request-name-field.md]: I was inconsistent on the very first AIP-131 one, and this is copy-pasted from that.

File: rules/aip0135/request_message_name.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	Name:   lint.NewRuleName("core", "0135", "request-message-name"),
+	URI:    "https://aip.dev/135#guidance",
+	OnlyIf: isDeleteMethod,
+	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
+		// Rule check: Establish that for methods such as `DeleteFoo`, the request
+		// message is named `DeleteFooRequest`.
+		if got, want := m.GetInputType().GetName(), m.GetName()+"Request"; got != want {
+			return []lint.Problem{{
+				Message: fmt.Sprintf(
+					"Delete RPCs should have a request message named after the RPC, such as %q.",
```
[the remark this answers is not in the document]
Comment (lukesneeringer) [reply rules/aip0135/request_message_name.go]: The way to fix this is with suggestions, which are coming once I add a bunch of location helper functions. I would like to consider it out of scope for this PR.

[the remark this answers is not in the document]
Comment (robcapo) [reply rules/aip0135/request_message_name.go]: Not going to block on it I just think "should have a request message named after the RPC" is ambiguous as to what is acceptable.

File: docs/rules/core/0135-http-uri-name.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+---
+rule:
+  aip: 135
+  name: [core, '0135', http-uri-name]
+  summary: Delete methods must map the name field to the URI.
+---
+
+# Get methods: Request message
```
[the remark this answers is not in the document]
Comment (lukesneeringer) [reply docs/rules/core/0135-http-uri-name.md]: This entire header is wrong. Fixed for both 131 and 135.

File: docs/rules/core/0135-http-body.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    delete: "/v1/{name=publishers/*/books/*}"
+    body: "*"
+  };
+}
+```
+
+If you need to violate this rule for an entire file, place the comment at the
+top of the file.
+
+[aip-131]: https://aip.dev/131
```
[the remark this answers is not in the document]
Comment (lukesneeringer) [reply docs/rules/core/0135-http-body.md]: Yep, thanks. Fixed.

File: docs/rules/core/0135-request-name-field.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukesneeringer) [reply docs/rules/core/0135-request-name-field.md]: Fixed in both.

File: rules/aip0135/request_message_name.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (lukesneeringer) [reply rules/aip0135/request_message_name.go]: That seems reasonable, I just do not want to do it here. I promise I plan on going back through as I do the location helper work and improve this.

[the remark this answers is not in the document]
Comment (robcapo) [reply rules/aip0135/request_message_name.go]: Sorry, I meant to LGTM this. That's fine with me.

**Pull request merged**